mypy.ini

```diff
@@ -8,11 +8,5 @@ ignore_errors=True
 [mypy-pandas.core.indexes.datetimelike]
 ignore_errors=True
 
-[mypy-pandas.core.indexes.datetimes]
-ignore_errors=True
-
 [mypy-pandas.core.indexes.period]
```

Comment:
I believe `pandas.core.indexes.period` can be removed too from the blacklist with these changes. I tried removing it and `mypy` reports no errors.

Reply:
I will take a look, thanks for the heads up.

Reply:
Well there is just one error in `pandas.core.indexes.period`, but since this PR has been merged already, I created a new one #26518